aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::natives::{
+    aggregator_natives::helpers_v2::{
+        aggregator_snapshot_string_info, aggregator_snapshot_u128_info,
+        aggregator_snapshot_u64_info,
+    },
+    AccountAddress,
+};
+use aptos_gas_schedule::gas_params::natives::aptos_framework::*;
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeResult,
+};
+use aptos_types::vm_status::StatusCode;
+use move_binary_format::errors::PartialVMError;
+use move_core_types::value::{MoveStructLayout, MoveTypeLayout};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Struct, StructRef, Value},
+};
+use smallvec::{smallvec, SmallVec};
+use std::{collections::VecDeque, ops::Deref};
+
+/***************************************************************************************************
+ * native fun create_snapshot(value: Element): AggregatorSnapshot<Element>;
+ **************************************************************************************************/
+
+fn native_create_snapshot(
```

Comment:
we need to add a feature flag , and gate in rust code in native_create_snapshot?

Reply:
Done.